polygon/rest/models/request.py

```diff
@@ -0,0 +1,101 @@
+from typing import Dict, Optional
+
+X_POLYGON_EDGE_ID = "X-Polygon-Edge-ID"
+X_POLYGON_EDGE_IP_ADDRESS = "X-Polygon-Edge-IP-Address"
+X_POLYGON_EDGE_USER_AGENT = "X-Polygon-Edge-User-Agent"
+
+HEADER = "header"
+
+
+class RequestOptionBuilder:
+    def __init__(
+        self,
+        edge_id: Optional[str] = None,
+        edge_ip_address: Optional[str] = None,
+        edge_user: Optional[str] = None,
+    ):
+        """
+        RequestOptionBuilder is a utility class to build polygon api options used in requests.
+        :param edge_id: is a required Launchpad header. It identifies the Edge User requesting data
+        :param edge_ip_address: is a required Launchpad header. It denotes the originating IP Address of the Edge User
+        :param edge_user: is an optional Launchpad header. It denotes the originating UserAgent of the Edge User requesting data.
+        """
+        self.headers: Optional[Dict[str, str]] = None
+        if edge_id is not None and edge_ip_address is not None:
```

Comment:
What if either is specified? I think `or` should be used here

Reply:
both are required. Everything is optional so we can instantiate without including them. it's somewhat pre optimized/ gives the user a little more flexibility. 
in the future if we add more options this would be a good style for different defaults; it doesn't make as much sense now that we are only providing headers.